ui/src/app/base/components/TableConfirm/TableConfirm.tsx

```diff
@@ -0,0 +1,88 @@
+import type { ReactNode } from "react";
+
+import {
+  ActionButton,
+  Button,
+  Col,
+  Notification,
+  Row,
+} from "@canonical/react-components";
+import type { Props as ButtonProps } from "@canonical/react-components/dist/components/Button";
+
+import { COL_SIZES } from "app/base/constants";
+import { useCycled } from "app/base/hooks";
+
+export type Props = {
+  confirmAppearance?: ButtonProps["appearance"];
+  confirmLabel: string;
+  errors?: string | Record<string, string[]>;
+  errorKey?: string;
+  finished: boolean;
+  inProgress: boolean;
+  message: ReactNode;
+  onClose: () => void;
+  onConfirm: () => void;
+  sidebar?: boolean;
+};
+
+const TableConfirm = ({
```

Comment:
@ClementChaumel here's the new confirm component I added that fits in with the style of our delete confirmations and I also updated it to handle API errors.